core/src/services/foundationdb/backend.rs

```diff
@@ -0,0 +1,238 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::cmp::min;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use foundationdb::api::NetworkAutoStop;
+use foundationdb::tuple::Subspace;
+use foundationdb::{Database, FdbError, RangeOption};
+use lazy_static::lazy_static;
+
+use futures::StreamExt;
+use std::iter::Iterator;
+
+use crate::raw::adapters::kv;
+use crate::Builder;
+use crate::Error;
+use crate::ErrorKind;
+use crate::Scheme;
+use crate::*;
+
+lazy_static! {
+    static ref FOUNDATIONDB_API_INITIALIZE_HANDLE: NetworkAutoStop =
+        unsafe { foundationdb::boot() };
```

Comment:
What I was trying to do here is to ensure `unsafe { foundationdb::boot() }` is only called once within each process. This line initializes foundationdb API and after this line executes, OpenDAL users can connect to as many instances as they want. 
I searched but didn't figure out another proper way to do that. IMO the variable name is confusing and maybe I'll change it to `FOUNDATIONDB_API_INITIALIZED` and use `once_cell`. Do you have any other suggestions?

Reply:
> IMO the variable name is confusing and maybe I'll change it to `FOUNDATIONDB_API_INITIALIZED` and use `once_cell`.

I don't think we need to do this. It's ok to place it in `Adapter` since we make sure that the drop will be called. For example:

```rust
struct Adapter {
  _network: OnceCell<Xxxx>
}
```

Reply:
If we put it in `Adapter`, there will be compilation errors showing 
```
  _network: OnceCell<NetworkAutoStop>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `NetworkAutoStop`
```

As we do not need one `_network` for each instance, and only need once at the whole process, maybe changing to a static `Lazy<OnceCell<NetworkAutoStop>>` is more suitable.

Reply:
Negative. A lib like opendal should start a thread in a static which could lead our application can't join it on need. 

We can try add an Arc here.

Reply:
Thanks. I will try it out.

Reply:
Thank you for your understanding and effort.